serde/src/de/value.rs

```diff
@@ -35,7 +41,9 @@ pub enum Error {
 }
 
 impl de::Error for Error {
-    fn syntax(_: &str) -> Self { Error::SyntaxError }
+    fn syntax(msg: &str) -> Self { Error::SyntaxError(String::from(msg)) }
+    fn type_mismatch(_type: de::Type) -> Self { Error::TypeError(_type) }
```

Comment:
Could you use `type_` or `ty` instead of `_type` to make sure that it always gets used?